### ISSUE TITLE
Deeplearning4j Updater based on genetic algorithm

### ISSUE THREAD

**Issue by SchweinchenFuntik** [deeplearning4j/deeplearning4j#7595]
Can Deeplearning4j be used to train an artificial neural network using genetic algorithms ? Found that `IUpdate` returns a `GradientUpdater`

**Comment by AlexDBlack** [comment]
In principle, sure. We don't have an implementation of this, however. You have full access to the parameters to the network (via params() and similar), so you could add some sort of genetic algorithm on top... I wouldn't try to integrate this into the existing updater system however - you don't need gradients for GA, so there's no point calculating them in the first place.

In practice, we have little interest in supporting this. To the best of my knowledge, genetic algorithms aren't efficient enough to have any practical use in training neural networks, when compared to standard stochastic gradient descent approaches. Adding GA optimization isn't something we are considering doing.

**Comment by SchweinchenFuntik** [comment]
@AlexDBlack 
Sorry for bothering. I use your framework for research purposes. Do you mean the parameters of the finished model (`model.params()`)? Then can I use your builders?

**Comment by SchweinchenFuntik** [comment]
I correctly understood that I need to override `model.fit (input, output)` with a manual update of `weight` ?

**Comment by AlexDBlack** [comment]
> Do you mean the parameters of the finished model (model.params())? Then can I use your builders?

>I correctly understood that I need to override model.fit (input, output) with a manual update of weight ?

I mean construct the network as normal, using standard builder pattern.
Then (the way I imagined this working) is that instead of calling fit (or calculating gradients) you would implement the GA optimization in a utility method. You could modify MultiLayerNetwork/ComputationGraph (or add new methods) but it'd be doing exactly the same thing as the utility method (getting parameters, modifying using GA, evaluating, etc).

**Comment by SchweinchenFuntik** [comment]
@AlexDBlack thank you very much
